backend/lucene/src/main/java/org/hibernate/search/backend/lucene/search/query/impl/LuceneSearcherImpl.java

```diff
@@ -165,7 +165,7 @@ private LuceneCollectors buildCollectors(IndexSearcher indexSearcher, IndexReade
 	}
 
 	private int getMaxDocs(IndexReader reader, int offset, Integer limit) {
-		if ( limit == null ) {
+		if ( limit == null || (long) offset + limit >= Integer.MAX_VALUE ) {
```

Comment:
This is not in sync with https://github.com/hibernate/hibernate-search/pull/2774 . I'll just skip this commit and rebase on `main` once we merged https://github.com/hibernate/hibernate-search/pull/2774 .

Reply:
Yes it is only temporary. Good you didn't merge it!